File: practice16/14.cpp
//https://www.nowcoder.com/practice/2d47abf176f7486db0d707088404db8a?tpId=90&tqId=30883&tPage=6&rp=6&ru=/ta/2018test&qru=/ta/2018test/question-ranking
#include<iostream>
using namespace std;
int main()
{
	int i, j, m, n, sum, count;
	while (cin >> m >> n)
	{
		count = 0;
		int **pt = new int *[n];
		for (i = 0; i < n; i++)
		{
			pt[i] = new int[m]();
		}
		sum = 0, i = -1, j = -1;
		while (1)
		{
			for (i++, j++; j < m; j++)
			{
				if (pt[i][j] == 0)
				{
					pt[i][j] = sum;
					count++;
				}
				else
				{
					break;
				}
			}
			sum = sum + 1;
			for (i++, j--; i < n; i++)
			{
				if (pt[i][j] == 0)
				{
					pt[i][j] = sum;
					count++;
				}
				else
				{
					break;
				}
			}
			sum++;
			for (i--, j--; j >= 0; j--)
			{
				if (pt[i][j] == 0)
				{
					pt[i][j] = sum;
					count++;
				}
				else
				{
					break;
				}
			}
			sum++;
			for (i--, j++; i > 0; i--)
			{
				if (pt[i][j] == 0)
				{
					pt[i][j] = sum;
					count++;
				}
				else
				{
					break;
				}
			}
			sum++;
			if (count == m*n)
			{
				break;
			}
		}
		sum++;
		for (i = 0; i < n; i++)
		{
			for (j = 0; j < m; j++)
			{
				cout << pt[i][j];
			}
			cout << endl;
		}
		delete[]pt;
	}
	system("pause");
	return 0;
}

File: practice32/14.java

// https://www.nowcoder.com/practice/61cfbb2e62104bc8aa3da5d44d38a6ef?tpId=85&tqId=29856&tPage=2&rp=2&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        // 小于6的值
        if (n <= 5 || n % 2 != 0) {
            System.out.println(-1);
            System.exit(0);
        }
        // 其他
        if (n % 8 == 0) {
            System.out.println(n / 8);
        } else if ((n % 8) % 2 == 0 && n != 10) {
            System.out.println(n / 8 + 1);
        } else {
            System.out.println(-1);
        }
    }
}
